serpextract/serpextract.py

```diff
@@ -432,15 +432,15 @@ def parse(self, url_parts):
 
                 # Now we have to check for a tricky case where it is a SERP but
                 # there are no keywords
-                if keyword is None and (u'&{}='.format(extractor) in query or
-                                        u'?{}='.format(extractor) in query):
+                if not keyword and (u'&{}='.format(extractor) in original_query or
```

Comment:
Wait a minute... why don't we just do a `extractor in query` here instead or all this substring stuff? That will be faster than this substring checking, and this was probably a mistake on my part in the past.